Extension/src/Debugger/attachQuickPick.ts

```diff
@@ -30,44 +30,42 @@ export interface AttachItem extends vscode.QuickPickItem {
     id?: string;
 }
 
-export function showQuickPick(getAttachItems: () => Promise<AttachItem[]>): Promise<string | undefined> {
-    return getAttachItems().then(processEntries =>
-        new Promise<string>((resolve, reject) => {
-            const quickPick: vscode.QuickPick<AttachItem> = vscode.window.createQuickPick<AttachItem>();
-            quickPick.title = localize("attach.to.process", "Attach to process");
-            quickPick.canSelectMany = false;
-            quickPick.matchOnDescription = true;
-            quickPick.matchOnDetail = true;
-            quickPick.placeholder = localize("select.process.attach", "Select the process to attach to");
-            quickPick.items = processEntries;
-            quickPick.buttons = [new RefreshButton()];
-
-            const disposables: vscode.Disposable[] = [];
-
-            quickPick.onDidTriggerButton(button => {
-                getAttachItems().then(processEntries => quickPick.items = processEntries);
-            }, undefined, disposables);
-
-            quickPick.onDidAccept(() => {
-                if (quickPick.selectedItems.length !== 1) {
-                    reject(new Error(localize("process.not.selected", "Process not selected.")));
-                }
+export async function showQuickPick(getAttachItems: () => Promise<AttachItem[]>): Promise<string | undefined> {
+    const processEntries: AttachItem[] = await getAttachItems();
+    return new Promise<string | undefined>((resolve, reject) => {
+        const quickPick: vscode.QuickPick<AttachItem> = vscode.window.createQuickPick<AttachItem>();
+        quickPick.title = localize("attach.to.process", "Attach to process");
+        quickPick.canSelectMany = false;
+        quickPick.matchOnDescription = true;
+        quickPick.matchOnDetail = true;
+        quickPick.placeholder = localize("select.process.attach", "Select the process to attach to");
+        quickPick.items = processEntries;
+        quickPick.buttons = [new RefreshButton()];
+
+        const disposables: vscode.Disposable[] = [];
+
+        quickPick.onDidTriggerButton(async () => { quickPick.items = await getAttachItems(); });
```

Comment:
You need to add back in the `undefined` and `disposables` arguments to `onDidTriggerButton` so that the event listener can be cleaned up later.